src/Query/QueryBuilder.php

```diff
@@ -335,6 +335,13 @@ public function executeStatement(): int
      */
     public function execute()
     {
+        Deprecation::trigger(
+            'doctrine/dbal',
+            'https://github.com/doctrine/dbal/pull/4578',
+            'QueryBuilder::execute is deprecated, instead use QueryBuilder::executeQuery (for SELECT) or' .
```

Comment:
This could be moved under the `if` statement and made more specific (without the "or").

Reply:
Good idea, I pushed it

Reply:
You should run `vendor/bin/phpcbf`

Reply:
done